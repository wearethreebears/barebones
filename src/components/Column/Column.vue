<template>
  <component
    :is="tag"
    :data-component="$options.name"
    :class="part['component']"
    ><slot
  /></component>
</template>

<script lang="ts">
export default { name: "Column" };
</script>

<script setup lang="ts">
import type { PropType } from "vue";
import type { Layout } from "@barebones/types/tag";

import useComponent from "@barebones/composables/useComponent";

import { classProps } from "@barebones-local/Column/Column.classes";
import parts from "@barebones-local/Column/Column";

const props = defineProps({
  ...classProps,
  tag: {
    required: false,
    type: String as PropType<Layout>,
    default: "div",
  },
});

const { part } = useComponent(parts, props, classProps);
</script>
