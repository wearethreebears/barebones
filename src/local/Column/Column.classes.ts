import type { PropType } from "vue";
import type { ClassProps } from "@barebones/types/prop";

export const classProps: ClassProps = {
  span: {
    type: [String, Array] as PropType<string | string[]>,
    default: "default",
  },
};
