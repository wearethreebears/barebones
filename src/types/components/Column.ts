import type { Parts } from "@barebones/types/component";

export interface Column {
  parts: Parts;
}
