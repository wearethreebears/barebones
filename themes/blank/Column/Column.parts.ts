import type { Parts } from "@barebones/types/component";
import type { Column } from "@barebones/types/components/Column";

// Base Classes

export default function ColumnLocal(): Column {
  const parts: Parts = {
    component: {},
    },
  };

  return {
    parts,
  };
}
