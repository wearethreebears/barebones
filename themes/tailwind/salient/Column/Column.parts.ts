import type { Parts } from "@barebones/types/component";
import type { Column } from "@barebones/types/components/Column";

// Base Classes

export default function ColumnLocal(): Column {
  const parts: Parts = {
    component: {
      span: {
        default: ["col-span-12"],
        "1": ["col-span-1"],
        "2": ["col-span-2"],
        "3": ["col-span-3"],
        "4": ["col-span-4"],
        "5": ["col-span-5"],
        "6": ["col-span-6"],
        "7": ["col-span-7"],
        "8": ["col-span-8"],
        "9": ["col-span-9"],
        "10": ["col-span-10"],
        "11": ["col-span-11"],
        "12": ["col-span-12"],
        "sm-1": ["sm:col-span-1"],
        "sm-2": ["sm:col-span-2"],
        "sm-3": ["sm:col-span-3"],
        "sm-4": ["sm:col-span-4"],
        "sm-5": ["sm:col-span-5"],
        "sm-6": ["sm:col-span-6"],
        "sm-7": ["sm:col-span-7"],
        "sm-8": ["sm:col-span-8"],
        "sm-9": ["sm:col-span-9"],
        "sm-10": ["sm:col-span-10"],
        "sm-11": ["sm:col-span-11"],
        "sm-12": ["sm:col-span-12"],
        "md-1": ["md:col-span-1"],
        "md-2": ["md:col-span-2"],
        "md-3": ["md:col-span-3"],
        "md-4": ["md:col-span-4"],
        "md-5": ["md:col-span-5"],
        "md-6": ["md:col-span-6"],
        "md-7": ["md:col-span-7"],
        "md-8": ["md:col-span-8"],
        "md-9": ["md:col-span-9"],
        "md-10": ["md:col-span-10"],
        "md-11": ["md:col-span-11"],
        "md-12": ["md:col-span-12"],
        "lg-1": ["lg:col-span-1"],
        "lg-2": ["lg:col-span-2"],
        "lg-3": ["lg:col-span-3"],
        "lg-4": ["lg:col-span-4"],
        "lg-5": ["lg:col-span-5"],
        "lg-6": ["lg:col-span-6"],
        "lg-7": ["lg:col-span-7"],
        "lg-8": ["lg:col-span-8"],
        "lg-9": ["lg:col-span-9"],
        "lg-10": ["lg:col-span-10"],
        "lg-11": ["lg:col-span-11"],
        "lg-12": ["lg:col-span-12"],
        "xl-1": ["xl:col-span-1"],
        "xl-2": ["xl:col-span-2"],
        "xl-3": ["xl:col-span-3"],
        "xl-4": ["xl:col-span-4"],
        "xl-5": ["xl:col-span-5"],
        "xl-6": ["xl:col-span-6"],
        "xl-7": ["xl:col-span-7"],
        "xl-8": ["xl:col-span-8"],
        "xl-9": ["xl:col-span-9"],
        "xl-10": ["xl:col-span-10"],
        "xl-11": ["xl:col-span-11"],
        "xl-12": ["xl:col-span-12"],
        "2xl-1": ["2xl:col-span-1"],
        "2xl-2": ["2xl:col-span-2"],
        "2xl-3": ["2xl:col-span-3"],
        "2xl-4": ["2xl:col-span-4"],
        "2xl-5": ["2xl:col-span-5"],
        "2xl-6": ["2xl:col-span-6"],
        "2xl-7": ["2xl:col-span-7"],
        "2xl-8": ["2xl:col-span-8"],
        "2xl-9": ["2xl:col-span-9"],
        "2xl-10": ["2xl:col-span-10"],
        "2xl-11": ["2xl:col-span-11"],
        "2xl-12": ["2xl:col-span-12"],
      },
    },
  };

  return {
    parts,
  };
}
