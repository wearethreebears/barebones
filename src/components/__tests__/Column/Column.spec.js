import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";
import { allComponentPartClassesToDisplayCorrectly } from "@barebones/components/__tests__/helpers";
import Column from "@barebones/components/Column/Column.vue";

import ColumnLocal from "@barebones-local/Column/Column.parts";
const { parts } = ColumnLocal();


const defaultSlot = "Hello world";

allComponentPartClassesToDisplayCorrectly(Column, parts);

describe("Column", () => {
  let wrapper;
  beforeEach(async () => {
    wrapper = mount(Column, {
      slots: {
        default: defaultSlot
      }
    });
  })

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("displays the correct slot", () => {
    expect(wrapper.html()).contains(defaultSlot)
  });

});
