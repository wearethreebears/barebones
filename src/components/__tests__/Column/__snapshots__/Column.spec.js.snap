// Vitest Snapshot v1

exports[`Column > renders properly 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="col-span-12"
        data-component="BonesColumn"
        data-part="component"
      >
        
        Hello world
        
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "hasOwnProperty": [Function],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": {
          "_ctx": [Circular],
          "default": [Function],
        },
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "span": "default",
            "tag": "div",
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "part": ComputedRefImpl {
              "__v_isReadonly": true,
              "__v_isRef": true,
              "_cacheable": true,
              "_dirty": false,
              "_setter": [Function],
              "_value": {
                "component": [
                  "col-span-12",
                ],
              },
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    Set {
                      [Circular],
                    },
                    [Circular],
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
              "effect": ReactiveEffect {
                "active": true,
                "computed": [Circular],
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
            },
            "parts": {
              "component": {
                "span": {
                  "1": [
                    "col-span-1",
                  ],
                  "10": [
                    "col-span-10",
                  ],
                  "11": [
                    "col-span-11",
                  ],
                  "12": [
                    "col-span-12",
                  ],
                  "2": [
                    "col-span-2",
                  ],
                  "2xl-1": [
                    "2xl:col-span-1",
                  ],
                  "2xl-10": [
                    "2xl:col-span-10",
                  ],
                  "2xl-11": [
                    "2xl:col-span-11",
                  ],
                  "2xl-12": [
                    "2xl:col-span-12",
                  ],
                  "2xl-2": [
                    "2xl:col-span-2",
                  ],
                  "2xl-3": [
                    "2xl:col-span-3",
                  ],
                  "2xl-4": [
                    "2xl:col-span-4",
                  ],
                  "2xl-5": [
                    "2xl:col-span-5",
                  ],
                  "2xl-6": [
                    "2xl:col-span-6",
                  ],
                  "2xl-7": [
                    "2xl:col-span-7",
                  ],
                  "2xl-8": [
                    "2xl:col-span-8",
                  ],
                  "2xl-9": [
                    "2xl:col-span-9",
                  ],
                  "3": [
                    "col-span-3",
                  ],
                  "4": [
                    "col-span-4",
                  ],
                  "5": [
                    "col-span-5",
                  ],
                  "6": [
                    "col-span-6",
                  ],
                  "7": [
                    "col-span-7",
                  ],
                  "8": [
                    "col-span-8",
                  ],
                  "9": [
                    "col-span-9",
                  ],
                  "default": [
                    "col-span-12",
                  ],
                  "lg-1": [
                    "lg:col-span-1",
                  ],
                  "lg-10": [
                    "lg:col-span-10",
                  ],
                  "lg-11": [
                    "lg:col-span-11",
                  ],
                  "lg-12": [
                    "lg:col-span-12",
                  ],
                  "lg-2": [
                    "lg:col-span-2",
                  ],
                  "lg-3": [
                    "lg:col-span-3",
                  ],
                  "lg-4": [
                    "lg:col-span-4",
                  ],
                  "lg-5": [
                    "lg:col-span-5",
                  ],
                  "lg-6": [
                    "lg:col-span-6",
                  ],
                  "lg-7": [
                    "lg:col-span-7",
                  ],
                  "lg-8": [
                    "lg:col-span-8",
                  ],
                  "lg-9": [
                    "lg:col-span-9",
                  ],
                  "md-1": [
                    "md:col-span-1",
                  ],
                  "md-10": [
                    "md:col-span-10",
                  ],
                  "md-11": [
                    "md:col-span-11",
                  ],
                  "md-12": [
                    "md:col-span-12",
                  ],
                  "md-2": [
                    "md:col-span-2",
                  ],
                  "md-3": [
                    "md:col-span-3",
                  ],
                  "md-4": [
                    "md:col-span-4",
                  ],
                  "md-5": [
                    "md:col-span-5",
                  ],
                  "md-6": [
                    "md:col-span-6",
                  ],
                  "md-7": [
                    "md:col-span-7",
                  ],
                  "md-8": [
                    "md:col-span-8",
                  ],
                  "md-9": [
                    "md:col-span-9",
                  ],
                  "sm-1": [
                    "sm:col-span-1",
                  ],
                  "sm-10": [
                    "sm:col-span-10",
                  ],
                  "sm-11": [
                    "sm:col-span-11",
                  ],
                  "sm-12": [
                    "sm:col-span-12",
                  ],
                  "sm-2": [
                    "sm:col-span-2",
                  ],
                  "sm-3": [
                    "sm:col-span-3",
                  ],
                  "sm-4": [
                    "sm:col-span-4",
                  ],
                  "sm-5": [
                    "sm:col-span-5",
                  ],
                  "sm-6": [
                    "sm:col-span-6",
                  ],
                  "sm-7": [
                    "sm:col-span-7",
                  ],
                  "sm-8": [
                    "sm:col-span-8",
                  ],
                  "sm-9": [
                    "sm:col-span-9",
                  ],
                  "xl-1": [
                    "xl:col-span-1",
                  ],
                  "xl-10": [
                    "xl:col-span-10",
                  ],
                  "xl-11": [
                    "xl:col-span-11",
                  ],
                  "xl-12": [
                    "xl:col-span-12",
                  ],
                  "xl-2": [
                    "xl:col-span-2",
                  ],
                  "xl-3": [
                    "xl:col-span-3",
                  ],
                  "xl-4": [
                    "xl:col-span-4",
                  ],
                  "xl-5": [
                    "xl:col-span-5",
                  ],
                  "xl-6": [
                    "xl:col-span-6",
                  ],
                  "xl-7": [
                    "xl:col-span-7",
                  ],
                  "xl-8": [
                    "xl:col-span-8",
                  ],
                  "xl-9": [
                    "xl:col-span-9",
                  ],
                },
              },
            },
            "props": {
              "span": "default",
              "tag": "div",
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {
            "hasOwnProperty": [Function],
          },
          "exposed": {
            "hasOwnProperty": [Function],
          },
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "span": "default",
            "tag": "div",
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "span": {
                "0": false,
                "1": true,
                "default": "default",
                "type": [
                  [Function],
                  [Function],
                ],
              },
              "tag": {
                "0": false,
                "1": true,
                "default": "div",
                "required": false,
                "type": [Function],
              },
            },
            [
              "span",
              "tag",
            ],
          ],
          "provides": {},
          "proxy": {
            "span": "default",
            "tag": "div",
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "effects": [
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": true,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": {
                    "component": [
                      "col-span-12",
                    ],
                  },
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": [
                        Set {
                          [Circular],
                        },
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {
              "default": [Function],
            },
          },
          "setupState": {
            "part": {
              "component": [
                "col-span-12",
              ],
            },
            "parts": {
              "component": {
                "span": {
                  "1": [
                    "col-span-1",
                  ],
                  "10": [
                    "col-span-10",
                  ],
                  "11": [
                    "col-span-11",
                  ],
                  "12": [
                    "col-span-12",
                  ],
                  "2": [
                    "col-span-2",
                  ],
                  "2xl-1": [
                    "2xl:col-span-1",
                  ],
                  "2xl-10": [
                    "2xl:col-span-10",
                  ],
                  "2xl-11": [
                    "2xl:col-span-11",
                  ],
                  "2xl-12": [
                    "2xl:col-span-12",
                  ],
                  "2xl-2": [
                    "2xl:col-span-2",
                  ],
                  "2xl-3": [
                    "2xl:col-span-3",
                  ],
                  "2xl-4": [
                    "2xl:col-span-4",
                  ],
                  "2xl-5": [
                    "2xl:col-span-5",
                  ],
                  "2xl-6": [
                    "2xl:col-span-6",
                  ],
                  "2xl-7": [
                    "2xl:col-span-7",
                  ],
                  "2xl-8": [
                    "2xl:col-span-8",
                  ],
                  "2xl-9": [
                    "2xl:col-span-9",
                  ],
                  "3": [
                    "col-span-3",
                  ],
                  "4": [
                    "col-span-4",
                  ],
                  "5": [
                    "col-span-5",
                  ],
                  "6": [
                    "col-span-6",
                  ],
                  "7": [
                    "col-span-7",
                  ],
                  "8": [
                    "col-span-8",
                  ],
                  "9": [
                    "col-span-9",
                  ],
                  "default": [
                    "col-span-12",
                  ],
                  "lg-1": [
                    "lg:col-span-1",
                  ],
                  "lg-10": [
                    "lg:col-span-10",
                  ],
                  "lg-11": [
                    "lg:col-span-11",
                  ],
                  "lg-12": [
                    "lg:col-span-12",
                  ],
                  "lg-2": [
                    "lg:col-span-2",
                  ],
                  "lg-3": [
                    "lg:col-span-3",
                  ],
                  "lg-4": [
                    "lg:col-span-4",
                  ],
                  "lg-5": [
                    "lg:col-span-5",
                  ],
                  "lg-6": [
                    "lg:col-span-6",
                  ],
                  "lg-7": [
                    "lg:col-span-7",
                  ],
                  "lg-8": [
                    "lg:col-span-8",
                  ],
                  "lg-9": [
                    "lg:col-span-9",
                  ],
                  "md-1": [
                    "md:col-span-1",
                  ],
                  "md-10": [
                    "md:col-span-10",
                  ],
                  "md-11": [
                    "md:col-span-11",
                  ],
                  "md-12": [
                    "md:col-span-12",
                  ],
                  "md-2": [
                    "md:col-span-2",
                  ],
                  "md-3": [
                    "md:col-span-3",
                  ],
                  "md-4": [
                    "md:col-span-4",
                  ],
                  "md-5": [
                    "md:col-span-5",
                  ],
                  "md-6": [
                    "md:col-span-6",
                  ],
                  "md-7": [
                    "md:col-span-7",
                  ],
                  "md-8": [
                    "md:col-span-8",
                  ],
                  "md-9": [
                    "md:col-span-9",
                  ],
                  "sm-1": [
                    "sm:col-span-1",
                  ],
                  "sm-10": [
                    "sm:col-span-10",
                  ],
                  "sm-11": [
                    "sm:col-span-11",
                  ],
                  "sm-12": [
                    "sm:col-span-12",
                  ],
                  "sm-2": [
                    "sm:col-span-2",
                  ],
                  "sm-3": [
                    "sm:col-span-3",
                  ],
                  "sm-4": [
                    "sm:col-span-4",
                  ],
                  "sm-5": [
                    "sm:col-span-5",
                  ],
                  "sm-6": [
                    "sm:col-span-6",
                  ],
                  "sm-7": [
                    "sm:col-span-7",
                  ],
                  "sm-8": [
                    "sm:col-span-8",
                  ],
                  "sm-9": [
                    "sm:col-span-9",
                  ],
                  "xl-1": [
                    "xl:col-span-1",
                  ],
                  "xl-10": [
                    "xl:col-span-10",
                  ],
                  "xl-11": [
                    "xl:col-span-11",
                  ],
                  "xl-12": [
                    "xl:col-span-12",
                  ],
                  "xl-2": [
                    "xl:col-span-2",
                  ],
                  "xl-3": [
                    "xl:col-span-3",
                  ],
                  "xl-4": [
                    "xl:col-span-4",
                  ],
                  "xl-5": [
                    "xl:col-span-5",
                  ],
                  "xl-6": [
                    "xl:col-span-6",
                  ],
                  "xl-7": [
                    "xl:col-span-7",
                  ],
                  "xl-8": [
                    "xl:col-span-8",
                  ],
                  "xl-9": [
                    "xl:col-span-9",
                  ],
                },
              },
            },
            "props": {
              "span": "default",
              "tag": "div",
            },
          },
          "slots": {
            "default": [Function],
          },
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": ,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Hello world",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": Hello world,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": ,
                "key": "_default",
                "patchFlag": -2,
                "props": {
                  "key": "_default",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 16,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Fragment),
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": ,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Hello world",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": Hello world,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": ,
                "key": "_default",
                "patchFlag": -2,
                "props": {
                  "key": "_default",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 16,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Fragment),
              },
            ],
            "dynamicProps": [
              "data-component",
              "class",
            ],
            "el": <div
              class="col-span-12"
              data-component="BonesColumn"
              data-part="component"
            >
              
              Hello world
              
            </div>,
            "key": null,
            "patchFlag": 8,
            "props": {
              "class": "col-span-12",
              "data-component": "BonesColumn",
              "data-part": "component",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/Users/liamhall/Documents/web-development/applications/barebones/src/components/Column/Column.vue",
            "__hmrId": "22d27662",
            "components": {},
            "name": "BonesColumn",
            "props": {
              "span": {
                "0": false,
                "1": true,
                "default": "default",
                "type": [
                  [Function],
                  [Function],
                ],
              },
              "tag": {
                "0": false,
                "1": true,
                "default": "div",
                "required": false,
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 147,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="col-span-12"
          data-component="BonesColumn"
          data-part="component"
        >
          
          Hello world
          
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 36,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/Users/liamhall/Documents/web-development/applications/barebones/src/components/Column/Column.vue",
          "__hmrId": "22d27662",
          "components": {},
          "name": "BonesColumn",
          "props": {
            "span": {
              "0": false,
              "1": true,
              "default": "default",
              "type": [
                [Function],
                [Function],
              ],
            },
            "tag": {
              "0": false,
              "1": true,
              "default": "div",
              "required": false,
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 146,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="col-span-12"
          data-component="BonesColumn"
          data-part="component"
        >
          
          Hello world
          
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 73,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.33",
  },
  "__setProps": [Function],
  "componentVM": {
    "span": "default",
    "tag": "div",
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <div
    class="col-span-12"
    data-component="BonesColumn"
    data-part="component"
  >
    
    Hello world
    
  </div>,
}
`;
